tests/templates/test_state_preparations.py

```diff
@@ -359,6 +389,98 @@ def circuit(state_vector):
             state_vector = np.array([0, 2, 0, 0])
             circuit(state_vector)
 
+    # fmt: off
+    @pytest.mark.parametrize("state_vector, n_wires", [
+        ([.5,.5,.5,.5],2),
+        ([1, 0, 0, 0], 2),
+        ([0, 1, 0, 0], 2),
+        ([0, 0, 0, 1], 2),
+        ([0, 1, 0, 0, 0, 0, 0, 0], 3),
+        ([0, 0, 0, 0, 1, 0, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+        ([1/2, 0, 0, 0, 1/2, 1/2, 1/2, 0], 3),
+        ([1/3, 0, 0, 0, 2/3, 2/3, 0, 0], 3),
+        ([2/3, 0, 0, 0, 1/3, 0, 0, 2/3], 3),
+    ])
+    # fmt: on
+
+    def test_RZ_skipped(self, state_vector, n_wires):
+        """Tests whether the cascade of RZ gates is skipped.
+        Note that currently there are also inefficiencies in the number of CNOT gates in the RY cascade.
+        When these are updated, this test will become ineffective"""
+        n_CNOT = 0
+        for i in range(1, n_wires):
+            n_CNOT = n_CNOT + 2 ** (i)
```

Comment:
Is this what we're trying to do?
https://www.wolframalpha.com/input/?i=sum_i%3D0%5EM+2%5Ei

Reply:
Yes! Thanks, I'll make the change.